### PR TITLE
Make the worker poll a job and then load the appropriate Exporter/Importer

### PR DESCRIPTION
**Pull request by rtannenbaum**
Make the worker poll a job, load all TransferExtensions, and retrieve the proper Exporter and Importer based on the polled job's metadata.

TransferExtensions are loaded via the ServiceLoader.

Remove TransferServiceProvider and consolidate with TransferExtension.

I left a TODO to refactor WorkerImpl as the polling and processing are now separated.

For #172

### CODE REVIEW

**Comment by rtannenbaum** [comment]
Based on our discussion today, I started a fresh PR here rather than continuing with #174.

I'm going to close that one as all the comments & unformatted code are stale...

File: portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/Exporter.java [hunk 1]
```diff
@@ -22,6 +22,9 @@
 /** Exports data from a source service. */
 public interface Exporter<A extends AuthData, T extends DataModel> {
 
+  /** The key associated with this service. */
+  String getServiceId();
```
Comment (holachuy) [inline portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/Exporter.java]: Per off line conversation, let's revert this

File: portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/Importer.java [hunk 2]
```diff
@@ -21,6 +21,9 @@
 /** Imports data into a destination service. */
 public interface Importer<A extends AuthData, T extends DataModel> {
 
+  /** The key associated with this service. */
+  String getServiceId();
```
Comment (holachuy) [inline portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/Importer.java]: Per off line conversation, let's revert this

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerMain.java [hunk 3]
```diff
@@ -24,38 +25,62 @@
...
+        exporter = extension.getExporter(transferDataType);
+      } else if (extension.getServiceId().equals(importService)) {
+        extension.initialize(extensionContext);
+        Preconditions.checkState(
+            importer == null,
+            "Can only instantiate importer from one matching TransferExtension. Previously found :"
+                + importer.getClass() + ", and now found: " + extension.getClass());
+        importer = extension.getImporter(transferDataType);
+      }
     }
```
Comment (jimmarino) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerMain.java]: As a minor suggestion for a later PR I think lines 57-75 could potentially be shortened by streaming the extension collection, filtering, finding the first matching, and getting or throwing on the returned Optional.

This is slightly different than checking if more than one extension matches.

Comment (rtannenbaum) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerMain.java]: Thanks, I was actually thinking the same thing... I left a TODO for now

File: portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/Exporter.java (hunk 1 above)
Comment (rtannenbaum) [reply portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/Exporter.java]: Done

File: portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/Importer.java (hunk 2 above)
Comment (rtannenbaum) [reply portability-spi-transfer/src/main/java/org/dataportabilityproject/spi/transfer/provider/Importer.java]: Done

**Pull request merged**